airflow/contrib/hooks/gcp_dataflow_hook.py

```diff
@@ -0,0 +1,161 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import logging
+import select
+import subprocess
+import time
+import uuid
+
+from apiclient.discovery import build
+
+from airflow.contrib.hooks.gcp_api_base_hook import GoogleCloudBaseHook
+
+
+class _DataflowJob:
+    def _get_job_id_from_name(self):
+        jobs = self.dataflow.projects().jobs().list(
+            projectId=self.project_number
+        ).execute()
+        print(str(jobs))
+        for job in jobs['jobs']:
+            print(str(job))
+            if job['name'] == self.job_name:
+                self.job_id = job['id']
+                return job
+        return None
+
+    def _get_job(self):
+        if self.job_id is None:
+            job = self._get_job_id_from_name()
+        else:
+            job = self.dataflow.projects().jobs().get(projectId=self.project_number,
+                                                      jobId=self.job_id).execute()
+        if 'currentState' in job:
+            logging.info('Google Cloud DataFlow job %s is %s', str(job['name']),
+                         str(job['currentState']))
+        else:
+            logging.info('Google Cloud DataFlow with job_id %s has name %s', self.job_id,
+                         str(job['name']))
+        return job
+
+    def wait_for_done(self):
+        while True:
+            if 'currentState' in self.job:
+                print self.job['currentState']
+                if 'JOB_STATE_DONE' == self.job['currentState']:
+                    return True
+                elif 'JOB_STATE_FAILED' == self.job['currentState']:
+                    raise Exception("Google Cloud Dataflow job " + str(
+                        self.job['name']) + " has failed.")
+                elif 'JOB_STATE_CANCELLED' == self.job['currentState']:
+                    raise Exception("Google Cloud Dataflow job " + str(
+                        self.job['name']) + " was cancelled.")
+                elif 'JOB_STATE_RUNNING' == self.job['currentState']:
+                    time.sleep(10)
+                else:
+                    logging.debug(str(self.job))
+                    raise Exception("Google Cloud Dataflow job " + str(
+                        self.job['name']) + " was unknown state: " + str(
+                        self.job['currentState']))
+            else:
+                time.sleep(15)
+
+            self.job = self._get_job()
+
+    def get(self):
+        return self.job
+
+    def __init__(self, dataflow, project_number, name):
```

Comment:
Constructor is usually the first function in the class. Please, place it in the beginning.